types/src/main/scala/aqua/types/Type.scala

```diff
@@ -219,7 +219,9 @@ case class OptionType(element: Type) extends BoxType {
 }
 
 // Struct is an unordered collection of labelled types
-case class StructType(name: String, fields: NonEmptyMap[String, Type]) extends DataType {
+case class StructType(name: String, fieldsList: NonEmptyList[(String, Type)]) extends DataType {
```

Comment:
It is just a `LabelledProductType` with `name`. So we have no unsorted product after this change, and have two sorted labelled products. I suppose it should be done another way.